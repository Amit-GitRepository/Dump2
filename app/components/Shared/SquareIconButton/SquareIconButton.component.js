import Icon from '../Icon/Icon.component';
import noop from 'lodash/noop';
import PropTypes from 'prop-types';
import React, {Component} from 'react';
import styles from './SquareIconButton.style';
import Touchable from '../Touchable/Touchable.component';
import {colors} from '../../../themes/constants.styles';
import {ISText} from '../../Shared';
import {View} from 'react-native';

class SquareIconButton extends Component {
  render () {
    const {title, subtitle, iconName, onPress, iconColor} = this.props;
    return (
      <Touchable borderless={true} onPress={onPress} style={styles.container}>
        <View style={styles.wrapper}>
          <Icon name={iconName} style={[styles.icon, {color: iconColor}]}/>
          <ISText style={styles.title} type='BOLD'>{title}</ISText>
          <View style={styles.subtextContainer}>
            <ISText type={'REGULAR'} style={styles.subtitle} noOfLines={1} ellipsizeMode={'tail'}>{subtitle}</ISText>
          </View>
        </View>
      </Touchable>
    );
  }
}

SquareIconButton.defaultProps = {
  title: '',
  subtitle: '',
  iconName: null,
  onPress: noop,
  iconColor: colors.PRIMARY_ACTIONABLE
};

SquareIconButton.propTypes = {
  title: PropTypes.string,
  subtitle: PropTypes.string,
  iconName: PropTypes.string,
  onPress: PropTypes.func,
  iconColor: PropTypes.string
};

export default SquareIconButton;
