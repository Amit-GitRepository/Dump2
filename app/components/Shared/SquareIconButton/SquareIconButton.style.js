import {colors, fonts} from '../../../themes/constants.styles';
import {shadowScattered} from '../../../themes/application.styles';

export default {
  container: {
    flex: 1,
    paddingHorizontal: 5,
    borderRadius: 15
  },
  wrapper: {
    alignItems: 'center',
    paddingBottom: 2,
    paddingTop: 6,
    borderRadius: 8.5,
    backgroundColor: colors.PRIMARY_BG_TEXT_CONTRAST,
    elevation: 3,
    marginBottom: 4,
    ...shadowScattered
  },
  title: {
    color: colors.SECONDARY_BG_TEXT_CONTRAST,
    fontSize: fonts.FONT_SIZE_NORMAL
  },
  subtextContainer: {
    marginTop: -5,
    flexDirection: 'row-reverse',
    alignItems: 'flex-end'
  },
  subtitle: {
    color: colors.SECONDARY_BG_TEXT_CONTRAST,
    fontSize: fonts.FONT_SIZE_SMALL
  },
  subIcon: {
    color: colors.PRIMARY_ACTIONABLE,
    paddingBottom: 5
  },
  icon: {
    fontSize: 30,
    padding: 1,
    color: colors.PRIMARY_ACTIONABLE
  }
};
