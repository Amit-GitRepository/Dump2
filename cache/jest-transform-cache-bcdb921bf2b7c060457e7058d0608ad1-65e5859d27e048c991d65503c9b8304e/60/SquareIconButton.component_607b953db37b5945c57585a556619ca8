Object.defineProperty(exports,"__esModule",{value:true});var cov_1vs6sklvo0=function(){var path='/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/Shared/SquareIconButton/SquareIconButton.component.js',hash='1eefef5d905e795a7978d9343193dae39d378116',global=new Function('return this')(),gcv='__coverage__',coverageData={path:'/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/Shared/SquareIconButton/SquareIconButton.component.js',statementMap:{'0':{start:{line:11,column:40},end:{line:11,column:50}},'1':{start:{line:12,column:4},end:{line:23,column:6}},'2':{start:{line:27,column:0},end:{line:31,column:2}},'3':{start:{line:33,column:0},end:{line:37,column:2}}},fnMap:{'0':{name:'(anonymous_0)',decl:{start:{line:10,column:2},end:{line:10,column:3}},loc:{start:{line:10,column:12},end:{line:24,column:3}},line:10}},branchMap:{},s:{'0':0,'1':0,'2':0,'3':0},f:{'0':0},b:{},_coverageSchema:'332fd63041d2c1bcb487cc26dd0d5f7d97098a6c'},coverage=global[gcv]||(global[gcv]={});if(coverage[path]&&coverage[path].hash===hash){return coverage[path];}coverageData.hash=hash;return coverage[path]=coverageData;}();var _createClass=function(){function defineProperties(target,props){for(var i=0;i<props.length;i++){var descriptor=props[i];descriptor.enumerable=descriptor.enumerable||false;descriptor.configurable=true;if("value"in descriptor)descriptor.writable=true;Object.defineProperty(target,descriptor.key,descriptor);}}return function(Constructor,protoProps,staticProps){if(protoProps)defineProperties(Constructor.prototype,protoProps);if(staticProps)defineProperties(Constructor,staticProps);return Constructor;};}();var _Icon=require('../Icon/Icon.component');var _Icon2=_interopRequireDefault(_Icon);
var _ISText=require('../../Shared/ISText/ISText.component');var _ISText2=_interopRequireDefault(_ISText);
var _propTypes=require('prop-types');var _propTypes2=_interopRequireDefault(_propTypes);
var _react=require('react');var _react2=_interopRequireDefault(_react);
var _SquareIconButton=require('./SquareIconButton.style');var _SquareIconButton2=_interopRequireDefault(_SquareIconButton);
var _Touchable=require('../Touchable/Touchable.component');var _Touchable2=_interopRequireDefault(_Touchable);
var _reactNative=require('react-native');function _interopRequireDefault(obj){return obj&&obj.__esModule?obj:{default:obj};}function _classCallCheck(instance,Constructor){if(!(instance instanceof Constructor)){throw new TypeError("Cannot call a class as a function");}}function _possibleConstructorReturn(self,call){if(!self){throw new ReferenceError("this hasn't been initialised - super() hasn't been called");}return call&&(typeof call==="object"||typeof call==="function")?call:self;}function _inherits(subClass,superClass){if(typeof superClass!=="function"&&superClass!==null){throw new TypeError("Super expression must either be null or a function, not "+typeof superClass);}subClass.prototype=Object.create(superClass&&superClass.prototype,{constructor:{value:subClass,enumerable:false,writable:true,configurable:true}});if(superClass)Object.setPrototypeOf?Object.setPrototypeOf(subClass,superClass):subClass.__proto__=superClass;}var

SquareIconButton=function(_Component){_inherits(SquareIconButton,_Component);function SquareIconButton(){_classCallCheck(this,SquareIconButton);return _possibleConstructorReturn(this,(SquareIconButton.__proto__||Object.getPrototypeOf(SquareIconButton)).apply(this,arguments));}_createClass(SquareIconButton,[{key:'render',value:function render()
{cov_1vs6sklvo0.f[0]++;var _ref=(cov_1vs6sklvo0.s[0]++,
this.props),title=_ref.title,subtitle=_ref.subtitle,iconName=_ref.iconName;cov_1vs6sklvo0.s[1]++;
return(
_react2.default.createElement(_Touchable2.default,{style:_SquareIconButton2.default.container},
_react2.default.createElement(_reactNative.View,{style:_SquareIconButton2.default.wrapper},
_react2.default.createElement(_Icon2.default,{name:iconName,style:_SquareIconButton2.default.icon}),
_react2.default.createElement(_ISText2.default,{style:_SquareIconButton2.default.title,type:'BOLD'},title),
_react2.default.createElement(_reactNative.View,{style:_SquareIconButton2.default.subtextContainer},
_react2.default.createElement(_Icon2.default,{style:_SquareIconButton2.default.subIcon,size:10,transform:_SquareIconButton2.default.subIcon.transform,name:'chevron-right'}),
_react2.default.createElement(_ISText2.default,{type:'SEMIBOLD',style:_SquareIconButton2.default.subtitle,noOfLines:1,ellipsizeMode:'tail'},subtitle+' ')))));




}}]);return SquareIconButton;}(_react.Component);cov_1vs6sklvo0.s[2]++;


SquareIconButton.defaultProps={
title:'',
subtitle:'',
iconName:null};cov_1vs6sklvo0.s[3]++;


SquareIconButton.propTypes={
title:_propTypes2.default.string,
subtitle:_propTypes2.default.string,
iconName:_propTypes2.default.string};exports.default=


SquareIconButton;