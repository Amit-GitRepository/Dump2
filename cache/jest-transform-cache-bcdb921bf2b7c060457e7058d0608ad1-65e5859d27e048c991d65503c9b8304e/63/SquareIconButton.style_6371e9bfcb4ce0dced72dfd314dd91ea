Object.defineProperty(exports,"__esModule",{value:true});var cov_1d84hyhgiw=function(){var path='/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/Shared/SquareIconButton/SquareIconButton.style.js',hash='b023f7ce38bc74e81699d08c8410290193e5c2b2',global=new Function('return this')(),gcv='__coverage__',coverageData={path:'/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/Shared/SquareIconButton/SquareIconButton.style.js',statementMap:{},fnMap:{},branchMap:{},s:{},f:{},b:{},_coverageSchema:'332fd63041d2c1bcb487cc26dd0d5f7d97098a6c'},coverage=global[gcv]||(global[gcv]={});if(coverage[path]&&coverage[path].hash===hash){return coverage[path];}coverageData.hash=hash;return coverage[path]=coverageData;}();var _constants=require('../../../themes/constants.styles');exports.default=

{
container:{
flex:1,
paddingHorizontal:5},

wrapper:{
alignItems:'center',
paddingVertical:5,
borderColor:_constants.colors.LANDING_TITLE_BG,
borderWidth:1.5,
borderRadius:15},

title:{
color:_constants.colors.SECONDARY_BG_TEXT_CONTRAST,
fontSize:_constants.fonts.FONT_SIZE_NORMAL},

subtextContainer:{
marginTop:-5,
flexDirection:'row-reverse',
alignItems:'flex-end'},

subtitle:{
color:_constants.colors.SECONDARY_BG_TEXT_CONTRAST,
fontSize:_constants.fonts.FONT_SIZE_SMALL},

subIcon:{
color:_constants.colors.PRIMARY_ACTIONABLE,
paddingBottom:5},

icon:{
fontSize:30,
padding:1,
color:_constants.colors.PRIMARY_ACTIONABLE}};