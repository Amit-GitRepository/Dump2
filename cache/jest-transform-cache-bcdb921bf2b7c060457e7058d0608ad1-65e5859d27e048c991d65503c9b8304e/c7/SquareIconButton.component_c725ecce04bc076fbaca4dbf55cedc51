Object.defineProperty(exports,"__esModule",{value:true});var cov_1vs6sklvo0=function(){var path='/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/Shared/SquareIconButton/SquareIconButton.component.js',hash='c239e2d6cfff22419d78ff1b45ae9d8c4467c392',global=new Function('return this')(),gcv='__coverage__',coverageData={path:'/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/Shared/SquareIconButton/SquareIconButton.component.js',statementMap:{'0':{start:{line:13,column:60},end:{line:13,column:70}},'1':{start:{line:14,column:4},end:{line:24,column:6}},'2':{start:{line:28,column:0},end:{line:34,column:2}},'3':{start:{line:36,column:0},end:{line:42,column:2}}},fnMap:{'0':{name:'(anonymous_0)',decl:{start:{line:12,column:2},end:{line:12,column:3}},loc:{start:{line:12,column:12},end:{line:25,column:3}},line:12}},branchMap:{},s:{'0':0,'1':0,'2':0,'3':0},f:{'0':0},b:{},_coverageSchema:'332fd63041d2c1bcb487cc26dd0d5f7d97098a6c'},coverage=global[gcv]||(global[gcv]={});if(coverage[path]&&coverage[path].hash===hash){return coverage[path];}coverageData.hash=hash;return coverage[path]=coverageData;}();var _createClass=function(){function defineProperties(target,props){for(var i=0;i<props.length;i++){var descriptor=props[i];descriptor.enumerable=descriptor.enumerable||false;descriptor.configurable=true;if("value"in descriptor)descriptor.writable=true;Object.defineProperty(target,descriptor.key,descriptor);}}return function(Constructor,protoProps,staticProps){if(protoProps)defineProperties(Constructor.prototype,protoProps);if(staticProps)defineProperties(Constructor,staticProps);return Constructor;};}();var _Icon=require('../Icon/Icon.component');var _Icon2=_interopRequireDefault(_Icon);
var _noop=require('lodash/noop');var _noop2=_interopRequireDefault(_noop);
var _propTypes=require('prop-types');var _propTypes2=_interopRequireDefault(_propTypes);
var _react=require('react');var _react2=_interopRequireDefault(_react);
var _SquareIconButton=require('./SquareIconButton.style');var _SquareIconButton2=_interopRequireDefault(_SquareIconButton);
var _Touchable=require('../Touchable/Touchable.component');var _Touchable2=_interopRequireDefault(_Touchable);
var _constants=require('../../../themes/constants.styles');
var _Shared=require('../../Shared');
var _reactNative=require('react-native');function _interopRequireDefault(obj){return obj&&obj.__esModule?obj:{default:obj};}function _classCallCheck(instance,Constructor){if(!(instance instanceof Constructor)){throw new TypeError("Cannot call a class as a function");}}function _possibleConstructorReturn(self,call){if(!self){throw new ReferenceError("this hasn't been initialised - super() hasn't been called");}return call&&(typeof call==="object"||typeof call==="function")?call:self;}function _inherits(subClass,superClass){if(typeof superClass!=="function"&&superClass!==null){throw new TypeError("Super expression must either be null or a function, not "+typeof superClass);}subClass.prototype=Object.create(superClass&&superClass.prototype,{constructor:{value:subClass,enumerable:false,writable:true,configurable:true}});if(superClass)Object.setPrototypeOf?Object.setPrototypeOf(subClass,superClass):subClass.__proto__=superClass;}var

SquareIconButton=function(_Component){_inherits(SquareIconButton,_Component);function SquareIconButton(){_classCallCheck(this,SquareIconButton);return _possibleConstructorReturn(this,(SquareIconButton.__proto__||Object.getPrototypeOf(SquareIconButton)).apply(this,arguments));}_createClass(SquareIconButton,[{key:'render',value:function render()
{cov_1vs6sklvo0.f[0]++;var _ref=(cov_1vs6sklvo0.s[0]++,
this.props),title=_ref.title,subtitle=_ref.subtitle,iconName=_ref.iconName,onPress=_ref.onPress,iconColor=_ref.iconColor;cov_1vs6sklvo0.s[1]++;
return(
_react2.default.createElement(_Touchable2.default,{onPress:onPress,style:_SquareIconButton2.default.container},
_react2.default.createElement(_reactNative.View,{style:_SquareIconButton2.default.wrapper},
_react2.default.createElement(_Icon2.default,{name:iconName,style:[_SquareIconButton2.default.icon,{color:iconColor}]}),
_react2.default.createElement(_Shared.ISText,{style:_SquareIconButton2.default.title,type:'BOLD'},title),
_react2.default.createElement(_reactNative.View,{style:_SquareIconButton2.default.subtextContainer},
_react2.default.createElement(_Shared.ISText,{type:'REGULAR',style:_SquareIconButton2.default.subtitle,noOfLines:1,ellipsizeMode:'tail'},subtitle)))));




}}]);return SquareIconButton;}(_react.Component);cov_1vs6sklvo0.s[2]++;


SquareIconButton.defaultProps={
title:'',
subtitle:'',
iconName:null,
onPress:_noop2.default,
iconColor:_constants.colors.PRIMARY_ACTIONABLE};cov_1vs6sklvo0.s[3]++;


SquareIconButton.propTypes={
title:_propTypes2.default.string,
subtitle:_propTypes2.default.string,
iconName:_propTypes2.default.string,
onPress:_propTypes2.default.func,
iconColor:_propTypes2.default.string};exports.default=


SquareIconButton;