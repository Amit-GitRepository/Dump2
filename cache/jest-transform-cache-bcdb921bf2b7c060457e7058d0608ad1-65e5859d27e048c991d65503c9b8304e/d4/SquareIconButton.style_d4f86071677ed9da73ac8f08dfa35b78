Object.defineProperty(exports,"__esModule",{value:true});var cov_1d84hyhgiw=function(){var path='/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/Shared/SquareIconButton/SquareIconButton.style.js',hash='b023f7ce38bc74e81699d08c8410290193e5c2b2',global=new Function('return this')(),gcv='__coverage__',coverageData={path:'/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/Shared/SquareIconButton/SquareIconButton.style.js',statementMap:{},fnMap:{},branchMap:{},s:{},f:{},b:{},_coverageSchema:'332fd63041d2c1bcb487cc26dd0d5f7d97098a6c'},coverage=global[gcv]||(global[gcv]={});if(coverage[path]&&coverage[path].hash===hash){return coverage[path];}coverageData.hash=hash;return coverage[path]=coverageData;}();var _extends=Object.assign||function(target){for(var i=1;i<arguments.length;i++){var source=arguments[i];for(var key in source){if(Object.prototype.hasOwnProperty.call(source,key)){target[key]=source[key];}}}return target;};var _constants=require('../../../themes/constants.styles');
var _application=require('../../../themes/application.styles');exports.default=

{
container:{
flex:1,
paddingHorizontal:5},

wrapper:_extends({
alignItems:'center',
paddingVertical:5,
borderRadius:15,
backgroundColor:_constants.colors.PRIMARY_BG_TEXT_CONTRAST,
elevation:3,
marginBottom:4},_application.shadowScattered),


title:{
color:_constants.colors.SECONDARY_BG_TEXT_CONTRAST,
fontSize:_constants.fonts.FONT_SIZE_NORMAL},

subtextContainer:{
marginTop:-5,
flexDirection:'row-reverse',
alignItems:'flex-end'},

subtitle:{
color:_constants.colors.SECONDARY_BG_TEXT_CONTRAST,
fontSize:_constants.fonts.FONT_SIZE_SMALL},

subIcon:{
color:_constants.colors.PRIMARY_ACTIONABLE,
paddingBottom:5},

icon:{
fontSize:30,
padding:1,
color:_constants.colors.PRIMARY_ACTIONABLE}};