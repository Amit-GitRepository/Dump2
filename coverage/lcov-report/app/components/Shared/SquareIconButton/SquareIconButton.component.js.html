<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for app/components/Shared/SquareIconButton/SquareIconButton.component.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../../index.html">All files</a> / <a href="index.html">app/components/Shared/SquareIconButton</a> SquareIconButton.component.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>4/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/1</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>4/4</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import Icon from '../Icon/Icon.component';
import ISText from '../../Shared/ISText/ISText.component';
import PropTypes from 'prop-types';
import React, {Component} from 'react';
import styles from './SquareIconButton.style';
import Touchable from '../Touchable/Touchable.component';
import {View} from 'react-native';
&nbsp;
class SquareIconButton extends Component {
  render () {
    const {title, subtitle, iconName} = this.props;
    return (
      &lt;Touchable style={styles.container}&gt;
        &lt;View style={styles.wrapper}&gt;
          &lt;Icon name={iconName} style={styles.icon}/&gt;
          &lt;ISText style={styles.title} type='BOLD' i18nKey={title} /&gt;
          &lt;View style={styles.subtextContainer}&gt;
            &lt;Icon style={styles.subIcon} size={10} transform={styles.subIcon.transform} name='chevron-right' /&gt;
            &lt;ISText type={'SEMIBOLD'} style={styles.subtitle} noOfLines={1} ellipsizeMode={'tail'} i18nKey= {subtitle} /&gt;
          &lt;/View&gt;
        &lt;/View&gt;
      &lt;/Touchable&gt;
    );
  }
}
&nbsp;
SquareIconButton.defaultProps = {
  title: '',
  subtitle: '',
  iconName: null
};
&nbsp;
SquareIconButton.propTypes = {
  title: PropTypes.string,
  subtitle: PropTypes.string,
  iconName: PropTypes.string
};
&nbsp;
export default SquareIconButton;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Tue Oct 31 2017 11:24:42 GMT+0530 (IST)
</div>
</div>
<script src="../../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../../sorter.js"></script>
</body>
</html>
