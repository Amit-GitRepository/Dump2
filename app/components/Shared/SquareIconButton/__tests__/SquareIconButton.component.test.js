import React from 'react';
import ReactSixteenAdapter from 'enzyme/build/adapters/ReactSixteenAdapter';
import renderer from 'react-test-renderer';
import SquareIconButton from '../SquareIconButton.component'; // Enzyme adapter for React 16
import {configure, shallow} from 'enzyme';

configure({adapter: new ReactSixteenAdapter()});

describe('SquareIconButton component', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<SquareIconButton title=''  subtitle=''  iconName='trueid'/>);
    const component = renderer.create(<SquareIconButton />).toJSON();
    expect(component).toMatchSnapshot();
    expect(wrapper).toBeDefined();
  });
});
