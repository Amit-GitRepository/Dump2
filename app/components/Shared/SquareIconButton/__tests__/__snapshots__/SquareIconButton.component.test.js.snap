// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SquareIconButton component renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "borderRadius": 15,
      "flex": 1,
      "opacity": 1,
      "paddingHorizontal": 5,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderRadius": 8.5,
          "elevation": 3,
          "marginBottom": 4,
          "paddingBottom": 2,
          "paddingTop": 6,
          "shadowColor": "#303C46",
          "shadowOffset": Object {
            "height": 2,
            "width": 1,
          },
          "shadowOpacity": 0.5,
          "shadowRadius": 2,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 12,
            },
            Array [
              Object {
                "color": "#F93E60",
                "fontSize": 30,
                "padding": 1,
              },
              Object {
                "color": "#F93E60",
              },
            ],
            Object {
              "fontFamily": "iservice",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "#000000",
            "fontFamily": "SukhumvitSet-Bold",
            "fontSize": 16,
          }
        }
      >
        
      </Text>
      <View
        style={
          Object {
            "alignItems": "flex-end",
            "flexDirection": "row-reverse",
            "marginTop": -5,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          noOfLines={1}
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "#000000",
              "fontFamily": "SukhumvitSet-Text",
              "fontSize": 14,
            }
          }
        >
          
        </Text>
      </View>
    </View>
  </View>
</View>
`;
